packages/ui-components/src/components/flow-canvas/canvas-context.tsx

```diff
@@ -0,0 +1,59 @@
+import { createContext, ReactNode, useContext, useState } from 'react';
+
+export type PanningMode = 'grab' | 'pan';
+
+type CanvasContextState = {
+  panningMode: PanningMode;
+  setPanningMode: (panningMode: PanningMode) => void;
+};
+
+const CanvasContext = createContext<CanvasContextState | undefined>(undefined);
+
+const DEFAULT_PANNING_MODE_KEY_IN_LOCAL_STORAGE = 'defaultPanningMode';
```

Comment:
this together with `const SHIFT_KEY = 'Shift';` could be extracted to a separate consts file